### PR TITLE
Properly revive processes on fork

### PR DESCRIPTION
**Pull request by rmg**
When upserting a ServiceProcess instance for a `fork` event, clear the `stopReason` and `stopTime` so that they are properly marked as running if they were previously marked as dead.

Also some small tweaks to some tests to work with an upgrade of tap.

### CODE REVIEW

**Comment by rmg** [comment]
@sam-github this is required for strongloop/strong-pm#189

**Comment by sam-github** [comment]
LGTM

but one thing that's not clear... when pm reconnects to a docker instance, it is going to get fork events for *all* living workers, right?

And also, the master process? I'm worried the children will be marked as alive, but the parent will still be marked 'stopped'.

**Comment by sam-github** [comment]
Also, all these tests are red, but they are all [green on master](http://ci.strongloop.com/job/strong-mesh-models/).

**Comment by rmg** [comment]
> but one thing that's not clear... when pm reconnects to a docker instance, it is going to get fork events for all living workers, right?

Yes, the docker driver will emit fake fork events for all living workers when it connects..

> And also, the master process? I'm worried the children will be marked as alive, but the parent will still be marked 'stopped'.

This is done by the `started` message already, but without this PR it results in creating a new "master" process each time instead of resurrecting the previous one when appropriate.

The code before this PR assumed that any process being "forked" is new and therefore has the default "null" values for stopReason/stopTime.

Basically this PR is making the findOrCreate-on-fork behaviour work as intended.

**Comment by rmg** [comment]
@sam-github had to expand the changes a bit, can you take another look before I merge?

**Comment by sam-github** [comment]
LGTM
